src/parser/sparqlParser/SparqlQleverVisitor.cpp

```diff
@@ -580,78 +580,198 @@ std::vector<ParsedQuery> Visitor::visit(Parser::UpdateContext* ctx) {
+  using Graph = SparqlTripleSimpleWithGraph::Graph;
+  return std::visit(
+      ad_utility::OverloadCallOperator{
+          [](const ad_utility::triple_component::Iri& iri) -> Graph {
+            return iri;
+          },
+          [](const ALL&) -> Graph { return Variable("?g"); },
+          [](const DEFAULT&) -> Graph { return std::monostate{}; },
+          [](const NAMED&) -> Graph { AD_FAIL(); }},
+      graph);
 }
 
 // ____________________________________________________________________________________
-Drop Visitor::visit(Parser::DropContext* ctx) {
-  return Drop{static_cast<bool>(ctx->SILENT()), visit(ctx->graphRefAll())};
+// Make a `GraphPatternOperation` that matches all triples in the graph.
+GraphPatternOperation makeAllTripleGraphPattern(const auto& graph) {
+  GraphPattern inner;
+  inner._graphPatterns.emplace_back(BasicGraphPattern{
+      {{{Variable("?s")}, Variable("?p"), {Variable("?o")}}}});
+  return {
+      parsedQuery::GroupGraphPattern{std::move(inner), transformGraph(graph)}};
 }
 
 // ____________________________________________________________________________________
-Create Visitor::visit(Parser::CreateContext* ctx) {
-  return Create{static_cast<bool>(ctx->SILENT()), visit(ctx->graphRef())};
+// Make a `SparqlTripleSimpleWithGraph` that templates all triples in the graph.
+SparqlTripleSimpleWithGraph makeAllTripleTemplate(const auto& graph) {
+  return {{Variable("?s")},
+          {Variable("?p")},
+          {Variable("?o")},
+          transformGraph(graph)};
 }
+}  // namespace
 
 // ____________________________________________________________________________________
-Add Visitor::visit(Parser::AddContext* ctx) {
+ParsedQuery Visitor::visit(Parser::ClearContext* ctx) {
+  return makeClear(visit(ctx->graphRefAll()));
+}
+
+ParsedQuery Visitor::makeClear(const GraphOrDefault& graph) {
+  parsedQuery_._rootGraphPattern._graphPatterns.push_back(
+      makeAllTripleGraphPattern(graph));
+  parsedQuery_._clause = parsedQuery::UpdateClause{
+      GraphUpdate{{}, {makeAllTripleTemplate(graph)}}};
+  parsedQuery_.datasetClauses_ = activeDatasetClauses_;
+  return parsedQuery_;
+}
+
+ParsedQuery Visitor::makeClear(const GraphRefAll& graph) {
+  if (std::holds_alternative<NAMED>(graph)) {
+    // We select all graphs and then filter out the default graph, to get only
+    // the named graphs.
+    parsedQuery_._rootGraphPattern._graphPatterns.push_back(
+        makeAllTripleGraphPattern(GraphRefAll{ALL{}}));
+    auto e = SparqlExpressionPimpl{
+        createExpression<sparqlExpression::NotEqualExpression>(
+            std::make_unique<sparqlExpression::VariableExpression>(
+                Variable("?g")),
+            std::make_unique<sparqlExpression::IriExpression>(
+                TripleComponent::Iri::fromIriref(DEFAULT_GRAPH_IRI))),
+        absl::StrCat("?g != ", DEFAULT_GRAPH_IRI)};
+    parsedQuery_._rootGraphPattern._filters.emplace_back(std::move(e));
```

Comment:
Add a `TODO<joka921,qup> Extend the graph filtering s.t. we can exclude a single graph more efficiently`.